includes/admin/onboarding/steps/step-products.php

```diff
@@ -0,0 +1,198 @@
+		exit();
+	}
+
+	if ( ! current_user_can( 'edit_products' ) ) {
+		return;
+	}
+
+	$response = array( 'success' => false );
+
+	// Prepare product post details.
+	$product = array(
+		'post_title'  => wp_strip_all_tags( $_REQUEST['product_title'] ),
+		'post_status' => 'draft',
+		'post_type'   => 'download',
+	);
+
+	// Insert the product into the database.
+	$post_id = wp_insert_post( $product );
+	if ( $post_id ) {
+		$post = get_post( $post_id );
+
+		// Save meta fields.
+		edd_download_meta_box_fields_save( $post_id, $post );
+
+		// Set featured image.
+		if ( ! empty( $_REQUEST['product_image_id'] ) ) {
+			set_post_thumbnail( $post_id, absint( $_REQUEST['product_image_id'] ) );
+		}
+
+		$response['success']      = true;
+		$response['redirect_url'] = get_edit_post_link( $post_id );
+	}
+
+	wp_send_json( $response );
+	exit;
+}
+
+/**
+ * Get step view.
+ *
+ * @since 3.2
+ */
+function step_html() {
+	$currency_position  = edd_get_option( 'currency_position', 'before' );
+	ob_start();
+	?>
+	<form method="post" class="edd-onboarding__create-product-form">
+		<input type="hidden" name="_edd_product_type" value="0">
+		<table class="form-table" role="presentation">
+			<tbody>
+				<tr>
+					<th scope="row"><h3><?php echo esc_html_e( 'Product details', 'easy-digital-downloads' ); ?></h3></th>
+					<td><span alt="f223" class="edd-help-tip dashicons dashicons-editor-help" title="Explanation for basic product details."></span></td>
+				</tr>
+				<tr>
+					<th scope="row"><label for="product_title"><?php echo esc_html_e( 'Product Name', 'easy-digital-downloads' ); ?></label></th>
+					<td>
+						<input type="text" class="regular-text" id="product_title" name="product_title" required>
+					</td>
+				</tr>
+				<tr>
+					<th scope="row"><label for="product_image_url"><?php echo esc_html_e( 'Product Image', 'easy-digital-downloads' ); ?></label></th>
+					<td>
+						<div class="edd-onboarding__product-image-wrapper">
+							<input type="hidden" id="product_image_id" name="product_image_id">
+							<input type="text" class="regular-text" id="product_image_url" class="" name="product_image_url" data-attachment-id-field="#product_image_id" />
+							<button data-input="#product_image_url" data-uploader-title="<?php echo esc_attr( __( 'Set image', 'easy-digital-downloads' ) ); ?>" data-uploader-button-text="<?php echo esc_attr( __( 'Set image', 'easy-digital-downloads' ) ); ?>" class="button edd_settings_upload_button button-secondary"><?php esc_html_e( 'Set Image', 'easy-digital-downloads' ); ?></button>
+						</div>
+					</td>
+				</tr>
+				<tr>
+					<th scope="row"><?php echo apply_filters( 'edd_price_options_heading', __( 'Pricing Options', 'easy-digital-downloads' ) ); ?></th>
+					<td>
+						<input style="display:none;" type="checkbox" class="edd-form-group__input" name="_variable_pricing" id="edd_variable_pricing" value="1"/>
+						<div class="edd-onbaording__pricing-option-pill">
```

Comment:
```suggestion
						<div class="edd-onboarding__pricing-option-pill">
```
Not sure if there is styling on that but if there is the rule needs to be updated also.